test/unit/ECDSAStakeRegistryUnit.t.sol

```diff
@@ -827,13 +834,239 @@ contract ECDSAStakeRegistryTest is ECDSAStakeRegistrySetup {
+        address retrievedOperator = registry.getLatestOperatorForSigningKey(initialSigningKey);
+
+        // Verify that the retrieved operator is the zero address
+        assertEq(
+            retrievedOperator,
+            address(0),
+            "The retrieved operator should be the zero address for an old signing key"
+        );
+    }
+
+    function test_GetOperatorForSigningKeyAtBlock_OldKey() public {
+        // Create a new operator and signing keys
+        address operator = operator3;
+        address initialSigningKey = signer;
+        address updatedSigningKey = address(vm.addr(signerPk + 1));
+
+        ISignatureUtilsMixinTypes.SignatureWithSaltAndExpiry memory operatorSignature;
+
+        // Register operator with the initial signing key
+        vm.prank(operator);
+        registry.registerOperatorWithSignature(operatorSignature, initialSigningKey);
+        
+        // Save the current block number
+        uint256 initialBlock = block.number;
+        vm.roll(block.number + 10);
+
+        // Update the operator's signing key
+        vm.prank(operator);
+        registry.updateOperatorSigningKey(updatedSigningKey);
+
+        // Query the operator by the old signing key at the old block
+        address retrievedOperator = registry.getOperatorForSigningKeyAtBlock(initialSigningKey, initialBlock);
+
+        // Verify that the retrieved operator matches the registered operator
+        assertEq(
+            retrievedOperator,
+            operator,
+            "The retrieved operator should match the original operator when querying at the old block"
+        );
+    }
+
+    function test_RevertsWhen_DuplicateSigningKey() public {
+        // Create a new registry instance for this test to avoid conflicts with the setup
+        IStrategy mockStrategy = IStrategy(address(0x1234));
+        IECDSAStakeRegistryTypes.Quorum memory quorum = IECDSAStakeRegistryTypes.Quorum({
+            strategies: new IECDSAStakeRegistryTypes.StrategyParams[](1)
+        });
+        quorum.strategies[0] = 
+            IECDSAStakeRegistryTypes.StrategyParams({strategy: mockStrategy, multiplier: 10000});
+        ECDSAStakeRegistry newRegistry = new ECDSAStakeRegistry(IDelegationManager(address(mockDelegationManager)));
+        newRegistry.initialize(address(mockServiceManager), 100, quorum);
+        
+        // Create two different operators
+        address operator1Address = address(0xABCD);
+        address operator2Address = address(0xDCBA);
+        // Same signing key for both operators
+        address sharedSigningKey = address(0x1111);
+
+        ISignatureUtilsMixinTypes.SignatureWithSaltAndExpiry memory operatorSignature;
+
+        // Register first operator with the signing key
+        vm.prank(operator1Address);
+        newRegistry.registerOperatorWithSignature(operatorSignature, sharedSigningKey);
+
+        // Try to register second operator with the same signing key
```

Comment:
Yes, good check.
Can you add a test to show this conflict is not possible after an update as well?